internal/addon/options.go

```diff
@@ -105,13 +106,14 @@ func BuildOptions(addOnDeployment *addonapiv1alpha1.AddOnDeploymentConfig) (Opti
 			opts.UserWorkloads.Logs.SubscriptionChannel = keyvalue.Value
 		// Platform Observability Options
 		case KeyMetricsHubHostname:
-			url, err := url.Parse(keyvalue.Value)
+			val := keyvalue.Value
+			if !strings.HasPrefix(val, "http") {
+				val = "https://" + val
+			}
+			url, err := url.Parse(val)
 			if err != nil {
 				return opts, fmt.Errorf("%w: %s", errInvalidMetricsHubHostname, err.Error())
 			}
```

Comment:
This makes it so we don't need to change the test case and it covers another invalid configuration. Feel free to also add a test case that ends with `:` i.e `metrics.endpoint:` as this is also invalid
```suggestion
			}
			
			// Hostname validation:
			// - Check if host is empty
			// - Check for invalid hostname formats like ":"
			if strings.TrimSpace(url.Host) == "" || url.Host == ":" ||
				strings.HasPrefix(url.Host, ":") {
				return opts, fmt.Errorf("%w: invalid hostname format '%s'", errInvalidMetricsHubHostname, url.Host)
			}

```

Reply:
On my side I don't mind if it doesn't cover every case. There are other checks later in the code. In the end I want current failure cases to be covered, such as a domain without a scheme and a missing domain when metrics collection is enabled. Which are both covered. 
Adding these additional checks and reverting the change on the test case.